### ISSUE TITLE
decodeTag causes exception, which in turn causes the ESP to crash on sleep and appear as a powerup event

### ISSUE THREAD

**Issue by mikevanis** [interactionresearchstudio/RFIDBirdFeeder#6]
Core dump:
```
>>>stack>>>

ctx: cont
sp: 3ffffd50 end: 3fffffc0 offset: 01b0
3fffff00:  00000318 00000063 00000063 40100870  
3fffff10:  3ffe8eb6 3ffeed51 68000a0d 4020b125  
3fffff20:  3ffe8eb6 00000000 2e646568 4020b125  
3fffff30:  3ffe8e1c 00000000 4bc6a7f0 3ffef094  
3fffff40:  3fffdad0 3ffeed51 3ffeed51 40201971  
3fffff50:  40207d38 3ffeed51 3ffef054 4020315e  
3fffff60:  3fffdad0 0000000f 3ffeef60 40208304  
3fffff70:  3fffdad0 3ffeef60 3ffeef60 40208378  
3fffff80:  3fffdad0 3ffeef60 3ffeed58 3ffef094  
3fffff90:  3fffdad0 00000000 3ffef054 4020331c  
3fffffa0:  feefeffe 00000000 3ffef054 4020a220  
3fffffb0:  feefeffe feefeffe 3ffe8544 40101115  
<<<stack<<<
```
Decoded: 
```
Decoding stack results
0x40100870: umm_free_core(void*) at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/umm_malloc/umm_malloc.cpp line 316
0x4020b125: uart_write(uart_t*, char const*, size_t) at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/uart.cpp line 509
0x4020b125: uart_write(uart_t*, char const*, size_t) at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/uart.cpp line 509
0x40201971: scanForTag(unsigned char*) at /var/folders/4k/6h5l9jl178sb4hslwv3xp1mr0000gn/T/arduino_modified_sketch_478001/rfid.ino line 248
0x40207d38: HardwareSerial::write(unsigned char const*, unsigned int) at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/HardwareSerial.h line 164
0x4020315e: updateRfid() at /var/folders/4k/6h5l9jl178sb4hslwv3xp1mr0000gn/T/arduino_modified_sketch_478001/RFIDUtilities.ino line 5
0x40208304: Print::println() at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/Print.cpp line 186
0x40208378: Print::println(char const*) at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/Print.cpp line 198
0x4020331c: loop() at /var/folders/4k/6h5l9jl178sb4hslwv3xp1mr0000gn/T/arduino_modified_sketch_478001/RFIDBirdFeeder.ino line 175
0x4020a220: loop_wrapper() at /Users/mikevanis/Library/Arduino15/packages/esp8266/hardware/esp8266/2.6.3/cores/esp8266/core_esp8266_main.cpp line 180
```
Commenting out line 247 in rfid.ino
```
tagCheck = decodeTag(tagData);
```
seems to stop it from crashing. Will have to investigate further, just opening the issue here so that I don't forget it.

**Comment by andysheen** [comment]
seems to be fixed in with commit fac8cf7f6df5d10e8a8b0d47169e2a68a746e306

a mixture of adding yield and extending the decoding time seemed to help it. 

I've also removed reference to using static declared variables dynamically(?!)

**Comment by mikevanis** [comment]
Nice one. Yes, that static declaration was so weird...!
